C++ API parity: AdaptiveAvgPool2d

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26915 C++ API parity: MaxUnpool2d
* #26896 C++ API parity: MaxUnpool1d
* #26819 C++ API parity: AdaptiveAvgPool3d
* **#26818 C++ API parity: AdaptiveAvgPool2d**

Differential Revision: [D17627822](https://our.internmc.facebook.com/intern/diff/D17627822)

pbelevich committed:
C++ API parity: AdaptiveAvgPool2d

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool2d"




[ghstack-poisoned]

yf225 commented on test/cpp/api/modules.cpp:
```diff
@@ -426,6 +426,50 @@ TEST_F(ModulesTest, AdaptiveAvgPool1d) {
+     {34.5, 36.5}},
+  }, torch::kFloat)));
+  ASSERT_EQ(y.sizes(), torch::IntArrayRef({2, 3, 2}));
+}
```
We need to test pretty print for `AdaptiveAvgPool2d` as well.

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool2d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool2d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool2d"




[ghstack-poisoned]

yf225 reviewed (approved):
Thanks @pbelevich !

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool2d"


Differential Revision: [D17627822](https://our.internmc.facebook.com/intern/diff/D17627822)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool2d"


Differential Revision: [D17627822](https://our.internmc.facebook.com/intern/diff/D17627822)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool2d"


Differential Revision: [D17627822](https://our.internmc.facebook.com/intern/diff/D17627822)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool2d"


Differential Revision: [D17627822](https://our.internmc.facebook.com/intern/diff/D17627822)

[ghstack-poisoned]
